superduperdb/base/serializable.py

```diff
@@ -48,105 +48,81 @@ def _deserialize(r: t.Any, db: None = None) -> t.Any:
     return instance
 
 
-def _serialize(item: t.Any) -> t.Dict[str, t.Any]:
-    def unpack(k, v):
-        attr = getattr(item, k)
-        if isinstance(attr, Serializable):
-            return _serialize(attr)
-
-        if isinstance(attr, (list, tuple)):
-            if isinstance(attr, tuple):
-                v = list(v)
-
-            for i, sc in enumerate(attr):
-                if isinstance(sc, Serializable):
-                    v[i] = _serialize(sc)
-
-        if isinstance(attr, dict):
-            for key, value in attr.items():
-                if isinstance(value, Serializable):
-                    v[key] = _serialize(value)
-        return v
-
-    d = {k: unpack(k, v) for k, v in item.dict().items()}
-
-    from superduperdb.components.component import Component
-
-    to_add = {}
-    if isinstance(item, Component):
-        to_add = {
-            'type_id': item.type_id,
-            'identifier': item.identifier,
-            'version': getattr(item, 'version', None),
+def _serialize(item: t.Any, serialize_variables: bool = True):
```

Comment:
This function was not flexible enough and was overly complicated for what it did. 
This should make it easier to understand.